### DISCUSSION TITLE
Intershop PWA for beginners

### DISCUSSION THREAD

**Discussion by jometzner** [intershop/intershop-dev-community#10] [Ideas]
We've talked about PWA in previous talks, but all of these assumed a certain familiarity with the key concepts. This topic is a placeholder for ideas what could be of interest to Intershop PWA beginners.

To start with something here are mine:
1. PWA concepts for Responsive Starter Store developers
2. Avoid these common mistakes

**Comment by MaxKless** [comment]
Sounds like a great idea. There's so many topics that this might have to be a series of talks, though... 😛 
How about an overview of PWA tooling, conventions and frameworks that you will need to know about when working with the PWA?

Tooling and Conventions: 
- "npm run ..." scripts available for the PWA
-  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) and [husky](https://typicode.github.io/husky/#/) commit hooks
- "npm run test" and what we consider good code
- The Github Actions pipeline and how to read it

Frameworks:
- [Jest](https://github.com/intershop/intershop-pwa/blob/develop/docs/guides/testing-jest.md) basics and how we use it
- [Cypress](https://github.com/intershop/intershop-pwa/blob/develop/docs/guides/testing-cypress.md) basics and how we use it (page object pattern)
- [Ngrx store](https://github.com/intershop/intershop-pwa/blob/develop/docs/guides/state-management.md) basics and how we use it
- [Formly ](https://github.com/intershop/intershop-pwa/blob/develop/docs/guides/formly.md) basics and how we use it

Most of this is documented, of course, but a talk could still very much be worthwhile.

**Comment by MaxKless** [comment]
@dhhyi could probably hold a very illuminating talk on product contexts that many, many people would be interested in 🥇
